modules/picard/filtersamreads/main.nf

```diff
@@ -0,0 +1,65 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process PICARD_FILTERSAMREADS {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::picard=2.25.6" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/picard:2.25.6--hdfd78af_0"
+    } else {
+        container "quay.io/biocontainers/picard:2.25.6--hdfd78af_0"
+    }
+
+    input:
+    tuple val(meta), path(bam)
+    val filter
+    path readlist
+
+    output:
+    tuple val(meta), path("*.filtered.bam"), emit: bam
+    path "*.version.txt"                   , emit: version
```

Comment:
```suggestion
    tuple val(meta), path("*.bam"), emit: bam
    path "*.version.txt"          , emit: version
```

Reply:
Wouldn't this also pick up the input `bam`?

Reply:
Yup. But you need to change the filename via `suffix`. We need to change this too https://github.com/nf-core/modules/blob/f43778b0e641a9643ae73fe1f2fe2d5bb81edffc/modules/picard/sortsam/main.nf#L44. Missed it.

Reply:
Like https://github.com/nf-core/modules/pull/602/files#r673951434

Reply:
This allows developers to call their output files whatever they want.

Reply:
May be worth fixing the `picard/sortsam` module first :)